File: arrays/containerWithMostWater.js
var maxArea = function (height) {
  let p1 = 0,
    p2 = height.length - 1,
    maxArea = 0;
  while (p1 < p2) {
    const heights = Math.min(height[p1], height[p2]);
    const width = p2 - p1;
    const area = heights * width;
    maxArea = Math.max(maxArea, area);
    if (height[p1] <= height[p2]) {
      p1++;
    } else {
      p2--;
    }
  }
  return maxArea;
};

File: arrays/container-with-most-water.py
class Solution(object):
    def maxArea(self, height):
        p1, p2 = 0, len(height) - 1
        max_area = 0
        while p1 < p2:
            heights = min(height[p1], height[p2])
            width = p2 - p1
            area = heights * width
            max_area = max(max_area,area)
            if height[p1] <= height[p2]:
                p1 += 1
            else:
                p2 -= 1
        return max_area